features/step_definitions/user_steps.rb

```diff
@@ -151,22 +151,12 @@ def page_heading_text(page)
   click_on('Register')
 end
 
-Given('they login as {string}') do |username|
-  fill_in('username', with: username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
-
-Given('they login as the newly registered user') do
-  fill_in('username', with: @username)
-  fill_in('password', with: 'bar')
-  click_on('SignIn')
-  click_on('I Agree')
-end
+Given /they login as "(.*)"(?: with a random pid|)$/ do |userstring|
```

Comment:
The optional `with a random pid` doesn't modify the behaviour in any way. Can it just be dropped, and also removed from the scenarios? So everything just becomes `they login as "some username"`?

Reply:
Ignore the above - I didn't see line 163. You probably need to capture if `with a random pid` is se though, and only run that line if it is.

Reply:
@Wynndow Changes made. Can you please re-review.